remoteworkitem/remoteworkitem_test.go

```diff
@@ -72,163 +65,299 @@ func TestWorkItemMapping(t *testing.T) {
 	assert.NotNil(t, workItem.Fields[models.SystemTitle], fmt.Sprintf("%s not mapped", models.SystemTitle))
 }
 
+// Table driven tests for the Mapping of Github issues
 func TestGitHubIssueMapping(t *testing.T) {
 	resource.Require(t, resource.UnitTest)
 
-	content, err := test.LoadTestData("github_issue_mapping.json", provideRemoteGithubDataWithAssignee)
-	if err != nil {
-		t.Fatal(err)
+	// githubData struct define test file and test url
+	type githubData struct {
+		inputFile      string
+		expectedOutput bool
+		inputURL       string
 	}
 
-	workItemMap := WorkItemKeyMaps[ProviderGithub]
-	remoteTrackerkItem := TrackerItem{Item: string(content[:]), RemoteItemID: "xyz", TrackerID: uint64(0)}
+	var gitData = []githubData{
+		// JSON data file of Github issue with assignee to test that the data is getting correctly Mapped through the Map funtion
+		// Github Issue API URL for the respective JSON data file to update the cache
+		{"github_issue_mapping.json", true, "http://api.github.com/repos/almighty-test/almighty-test-unit/issues/2"},
+		// JSON data file of Github issue with assignee and label
+		// Issue API URL for the respective JSON file to update the cache
+		{"github_test2_data.json", true, "https://api.github.com/repos/almighty-unit-test/almighty-test/issues/1"},
```

Comment:
Maybe give the files a name that represents something around their usage/test scenario?